src/nevermined/api/AssetsApi.ts

```diff
@@ -0,0 +1,464 @@
+    }
+
+    /**
+     * Registers a new asset in Nevermined. 
+     * You can find more information about how different data is stored in Nevermined here:
+     * {@link https://docs.nevermined.io/docs/architecture/nevermined-data}
+     *
+     * @param assetAttributes - Attributes describing the asset
+     * @param publishMetadata - Allows to specify if the metadata should be stored in different backends
+     * @param publisher - The account publishing the asset
+     * @param txParams - Optional transaction parameters
+     * @returns The metadata of the asset created (DDO)
+     *
+     * @returns {@link DDO}
+     */
+    public create(
+        assetAttributes: AssetAttributes,
+        publisher: Account,
+        publishMetadata: PublishMetadata = PublishMetadata.OnlyMetadataAPI,
+        txParams?: TxParameters
+    ): SubscribablePromise<CreateProgressStep, DDO> {
+        return this.registerNeverminedAsset(
+            assetAttributes,
+            publisher,
+            publishMetadata,
+            undefined,
+            txParams
+        )
+    }
+
+
+    /**
+     * Given a DID, updates the metadata associated to the asset. It also can upload this metadata to a remote decentralized stored depending on the `publishMetadata` parameter.
+     * 
+     * @example
+     * ```ts
+     * const ddoUpdated = await nevermined.assets.update(
+     *      ddo.shortId(), 
+     *      updatedMetadata, 
+     *      publisher, 
+     *      PublishMetadata.IPFS
+     * )
+     * ```
+     * 
+     * @param did - Decentralized ID representing the unique id of an asset in a Nevermined network.
+     * @param metadata - Metadata describing the asset
+     * @param publisher - Account of the user updating the metadata
+     * @param publishMetadata - It allows to specify where to store the metadata  
+     * @param txParams - Optional transaction parameters
+     * @returns {@link DDO} The DDO updated
+     */
+     public update(
+        did: string,
+        metadata: MetaData,
+        publisher: Account,
+        publishMetadata: PublishMetadata = PublishMetadata.OnlyMetadataAPI,
+        txParams?: TxParameters
+    ): SubscribablePromise<UpdateProgressStep, DDO> {
+        return this.updateAsset(did, metadata, publisher, publishMetadata, txParams)
+    }
+
+
+    /**
+     * Start the purchase/order of an access service. Starts by signing the service agreement
+     * then sends the request to the publisher via the service endpoint (Node http service).
+     * If the access service to purchase is having associated some price, it will make the payment
+     * for that service.
+     * @param did - Unique identifier of the asset to order
+     * @param consumerAccount - The account of the user ordering the asset
+     * @param txParams - Optional transaction parameters
+     * @returns The agreement ID identifying the order
+     */
+     public order(
+        did: string,
+        consumerAccount: Account,
```

Comment:
either use `consumer` or change the parameters of other methods to `publisherAccount`

Reply:
changed to use the `*Account` in all the method signatures